src/nhs_grid.jl

```diff
@@ -203,10 +206,16 @@ end
 # Update only with neighbor coordinates
 function update_grid!(neighborhood_search::GridNeighborhoodSearch{NDIMS},
                       y::AbstractMatrix; parallelization_backend = y) where {NDIMS}
+    check_domain_bounds(neighborhood_search.cell_list, y,
+                        search_radius(neighborhood_search))
```

Comment:
In terms of performance, does it make sense to switch off this check for the update?

Reply:
You're right. It's slow.

Two options:
1. Make this parallel like so:
```julia
function check_domain_bounds(cell_list::FullGridCellList, y, search_radius)
    valid_ = [true]
    backend = KernelAbstractions.get_backend(y)
    valid = Adapt.adapt(backend, valid_)

    @threaded y for point in 1:size(y, 2)
        coords = extract_svector(y, Val(3), point)
        if any(isnan, coords)
            valid[1] = false
        end

        if any(coords .- search_radius .< cell_list.min_corner) || any(coords .+ search_radius .> cell_list.max_corner)
            valid[1] = false
        end
    end
    if !Vector(valid)[1]
        error("particle coordinates are outside the domain bounds of the cell list")
    end
end
```
The alternating update benchmark yielded:
7.136 ms with Polyester.jl without this check
8.497 ms with Polyester.jl with the check
20.356 ms with CUDA.jl without the check
27.648 ms with CUDA.jl with the check

2. Add a check inside the update loop/kernel. This is what is currently implemented in this PR. This approach does not have a measurable overhead, but has the disadvantage that errors are thrown inside a multithreaded loop. With Polyester.jl, threads are just stopped on errors, but no error message is displayed:
```julia
julia> x = zeros(Int, 5, 10)
5×10 Matrix{Int64}:
 0  0  0  0  0  0  0  0  0  0
 0  0  0  0  0  0  0  0  0  0
 0  0  0  0  0  0  0  0  0  0
 0  0  0  0  0  0  0  0  0  0
 0  0  0  0  0  0  0  0  0  0

julia> @batch for j in axes(x, 2)
           for i in 1:5
               if i > 3 && j in (2, 4, 9)
                   error("some error")
               end
               x[i, j] = 1
           end
       end

julia> x
5×10 Matrix{Int64}:
 1  1  1  1  1  1  1  1  1  1
 1  1  1  1  1  1  1  1  1  1
 1  1  1  1  1  1  1  1  1  1
 1  0  1  0  1  1  1  1  0  1
 1  0  1  0  1  1  1  1  0  1
```
So in this case, particles outside the domain bounds will cause unpredictable behavior.